Makefile

```diff
@@ -0,0 +1,58 @@
+# Copyright (c) 2018 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+GOCMD=go
+GOFMT=${GOCMD} fmt
+GOGET=${GOCMD} get
+GOGENERATE=${GOCMD} generate
+GOINSTALL=${GOCMD} install
+GOTEST=${GOCMD} test
```

Comment:
The script fails if GOPATH is not set. There is a default GOPATH that go uses, it is defined as `$HOME/go`

Since this Makefile depends on the existence of GOPATH, if GOPATH is not set, we should set it to the default.

Reply:
You can use `GOPATH?=$(shell go env GOPATH)` to set it to the Go's default only if not set.

Reply:
Done, and thanks!